### ISSUE TITLE
JVP incorrect in forward mode for `tf.math.sign` and `tf.experimental.numpy.sign`

### ISSUE THREAD

**Issue by drewshark** [tensorflow/tensorflow#60736]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

binary

### Tensorflow Version

tf-nightly

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 20.04

### Current Behaviour?

When I was trying to calculate jvp of `tf.math.sign` and `tf.experimental.numpy.sign` in forward mode, it gives `None` instead of a proper value.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

x = tf.constant([1], dtype=tf.float32)

for i in range(tf.size(x)):
  tangents = tf.constant([1.])
  with tf.autodiff.ForwardAccumulator(x, tangents) as acc:
    value = tf.math.sign(x)
  jvp_i = acc.jvp(value)
  print(jvp_i)
```

### Relevant log output

```shell
None
```
</details>

**Comment by tilakrayal** [comment]
@drewshark,
Could you please refer to the documentation of `tf.autodiff.ForwardAccumulator` where it was stated that **Forward mode** works best on functions with many outputs and few inputs. Since it does not hold on to intermediate activations, it is much more memory efficient than backprop where it is applicable.

Reverse mode is more attractive when computing gradients of a scalar-valued function with respect to many inputs (e.g. a neural network with many parameters and a scalar loss).

https://www.tensorflow.org/api_docs/python/tf/autodiff/ForwardAccumulator
Thank you!
